packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1936,8 +1981,11 @@ ValueResult<std::unique_ptr<Machine>> ArbCore::getMachineForSideload(
     // Check the cache
     {
         std::shared_lock<std::shared_mutex> lock(sideload_cache_mutex);
-        auto it = sideload_cache.find(block_number);
-        if (it != sideload_cache.end()) {
+        // Look for the first value after the value we want
```

Comment:
I think this should be the first value `<= block_number`

Reply:
It should be, but C++'s APIs are stupid here and don't provide such a method, so instead we go to the first value greater than the block number, and then go back one.

Reply:
Ahh, I read the code to fast. Makes sense